### PR TITLE
#5274 DynamicView: Properly resize mirror instances after construction

### PR DESCRIPTION
**Pull request by PhilMiller**
The constructor arguments set the new instance's capacity, but its size is initially small/0

### CODE REVIEW

**Comment by PhilMiller** [comment]
Discovered in the course of https://github.com/DARMA-tasking/magistrate/pull/261

**Review by crtrott — APPROVED** [review]

**Comment by crtrott** [comment]
hm maybe there should be a constructor which makes the DynamicView start out with a size?

**Comment by PhilMiller** [comment]
Might be useful, but I don't want to get into immediate design for the sake of a bug fix

**Comment by PhilMiller** [comment]
Test failure:

```
53: [ RUN      ] cuda.create_mirror_view_and_copy_dynamicview
53: Test failure: encountered unwanted events
53:   Types didn't match on arguments
53:   Found fence event
53: BeginFence { "DynamicView::resize_serial: Fence after copying chunks to the device", 33554433,94}
53: EndFence { 94}
53: /var/jenkins/workspace/Kokkos/containers/unit_tests/TestWithoutInitializing.hpp:663: Failure
53: Value of: success
53:   Actual: false
53: Expected: true
53: [  FAILED  ] cuda.create_mirror_view_and_copy_dynamicview (0 ms)
```

I need to fix the test's expectations to match the sad reality

File: containers/unit_tests/TestWithoutInitializing.hpp [hunk 1]
```diff
@@ -689,6 +694,14 @@ TEST(TEST_CATEGORY, create_mirror_no_init_dynamicview_view_ctor) {
                                Kokkos::DefaultExecutionSpace{}),
             host_view);
       },
+      [&](BeginFenceEvent event) {
+        if (event.descriptor().find("DynamicView::resize_serial: Fence after "
+                                    "copying chunks to the device") !=
+            std::string::npos)
+          return MatchDiagnostic{false};
```
Comment (masterleinad) [inline containers/unit_tests/TestWithoutInitializing.hpp:701]: We should probably introduce a version that accepts an execution space instance for the `deep_copy` later.

**Review by masterleinad — APPROVED** [review]

**Comment by masterleinad** [comment]
Would you mind still adding a test to check that the size is as expected (which failed previously)?

**Comment by dalg24** [comment]
> Would you mind still adding a test to check that the size is as expected (which failed previously)?

Sorry Daniel, I merged as you posted your comment

**Comment by masterleinad** [comment]
> Sorry Daniel, I merged as you posted your comment

That's fine we can add in a follow-up.

**Pull request merged**